truffle/src/main/java/org/jruby/truffle/core/string/StringNodes.java

```diff
@@ -1355,6 +1360,153 @@ private int prevCharHead(Encoding enc, byte[]bytes, int p, int s, int end) {
         }
     }
 
+    @Primitive(name = "string_scrub")
+    @ImportStatic(StringGuards.class)
+    public abstract static class ScrubNode extends PrimitiveArrayArgumentsNode {
+
+        @Child private YieldNode yieldNode = new YieldNode();
+        @Child private RopeNodes.MakeConcatNode makeConcatNode = RopeNodesFactory.MakeConcatNodeGen.create(null, null, null);
+
+        @Specialization(guards = { "!is7Bit(string)", "!isValidCodeRange(string)", "isAsciiCompatible(string)" })
```

Comment:
Replace `"!is7Bit(string)", "!isValidCodeRange(string)"` with `isBrokenCodeRange(string)`. We never have ropes with unknown code ranges floating around, so we don't need to worry about that case. When you eliminate the other two, broken code ranges are all that remain. Incidentally, I think it also clarifies the intent more.